language-support/hs/bindings/test/DA/Ledger/Sandbox.hs

```diff
@@ -7,42 +7,37 @@ module DA.Ledger.Sandbox ( -- Run a sandbox for testing on a dynamically selecte
     Sandbox(..),
     startSandbox,
     shutdownSandbox,
-    withSandbox,
+    withSandbox
     ) where
 
-import Control.Exception (bracket, evaluate, onException)
-import Control.Monad(when)
+import Control.Concurrent (threadDelay)
+import Control.Exception (bracket)
 import DA.Bazel.Runfiles(exe,locateRunfiles,mainWorkspace)
-import DA.Ledger (Port (..), unPort)
-import Data.List (isInfixOf)
-import Data.List.Extra(splitOn)
-import GHC.IO.Handle (Handle, hGetLine)
+import DA.Ledger (Port (..))
+import GHC.IO.Handle (Handle)
+import Safe (readMay)
+import System.IO.Extra (withTempFile)
 import System.FilePath((</>))
-import System.Process (CreateProcess (..), ProcessHandle, StdStream (CreatePipe), createProcess, getPid, interruptProcessGroupOf, proc, waitForProcess)
-import System.Time.Extra (Seconds, timeout)
+import System.Process (CreateProcess (..), ProcessHandle, createProcess, getPid, interruptProcessGroupOf, proc, waitForProcess)
 
 import DA.Ledger.Trace
 
 data SandboxSpec = SandboxSpec {dar :: String}
 
 data Sandbox = Sandbox { port :: Port, proh :: ProcessHandle }
 
-selectedPort :: Int
-selectedPort = 0 --dynamic port selection
 
-sandboxProcess :: SandboxSpec -> IO CreateProcess
-sandboxProcess SandboxSpec{dar} = do
+sandboxProcess :: SandboxSpec -> FilePath -> IO CreateProcess
+sandboxProcess SandboxSpec{dar} portFile = do
     binary <- locateRunfiles (mainWorkspace </> exe "ledger/sandbox/sandbox-binary")
-    pure $ proc binary [ dar, "--port", show selectedPort]
+    pure $ proc binary [ dar, "--port-file", portFile]
 
-startSandboxProcess :: SandboxSpec -> IO (ProcessHandle,Maybe Handle)
-startSandboxProcess spec = do
-    processRecord <- sandboxProcess spec
+startSandboxProcess :: SandboxSpec -> FilePath -> IO (ProcessHandle,Maybe Handle)
+startSandboxProcess spec portFile = do
+    processRecord <- sandboxProcess spec portFile
     (_,hOutOpt,_,proh) <-
         createProcess processRecord {
-        std_out = CreatePipe,
```

Comment:
Does this mean that we are now spamming stdout and stderr during tests? From a quick experiment it looks like sandbox is still very noisy even if you pass `--port-file`. Maybe just redirect stdout to `/dev/null`? (stderr seems to be clean by default afaict). `typed-process` has a cross-platform version of [/dev/null](https://hackage.haskell.org/package/typed-process-0.2.6.0/docs/System-Process-Typed.html#v:nullStream) that we could either steal or we could just switch to `typed-process` which generally seems like a saner API (we’ve already switched in other places due to issues with `process`).

Reply:
yes, these are captured by Bazel and put into `test.log`, which I thought was better to keep, but on the other hand these are mixed with test results - I can redirect them to `/dev/null`

Reply:
I often run tests via `bazel run` to be able to pass arguments and see the output as it goes so I would prefer if they are not too noisy.